src/lewis/core/simulation.py

```diff
@@ -185,7 +186,25 @@ def start(self):
         self._running = False
         self._started = False
 
-        self._adapters.disconnect()
+        self.log.info('Simulation has ended.')
+
+    def _start_control_server(self):
+        if self._control_server is not None and self._control_server_thread is None:
+            def control_server_loop():
+                self._control_server.start_server()
+
+                while not self._stop_commanded:
+                    self._control_server.process(blocking=True)
+
+                self.log.info('Stopped processing control server commands, ending thread.')
+
+            self._control_server_thread = Thread(target=control_server_loop)
+            self._control_server_thread.start()
+
+    def _stop_control_server(self):
+        if self._control_server_thread is not None:
+            self._control_server_thread.join()
```

Comment:
It might be good to add a timeout to this, to make sure it doesn't hang forever if something goes wrong.

Reply:
Good point. Together with the recv-timeout, this will make deadlocks on exiting very unlikely.